packages/core/src/lib/wait_for_remote_peer.ts

```diff
@@ -24,6 +24,7 @@ const log = debug("waku:wait-for-remote-peer");
  * remote nodes, **rejects** if the timeoutMs is reached.
  * @throws If passing a protocol that is not mounted
  * @default Wait for remote peers with protocols enabled locally and no time out is applied.
+ * @deprecated: this does not need to be called explicitly anymore, and is safe to completely remove from your implementation. The waiting for peer  logic is now built into the protocols themselves.
```

Comment:
I am not sure we want to fully deprecate it just yet and instead, might prefer to dog food the new strategy first.

Reply:
Yes I think we would need to do some dogfooding first and see how it goes when we use it in the examples.